synapse/storage/event_federation.py

```diff
@@ -203,15 +203,36 @@ def get_rooms_with_many_extremities(self, min_count, limit):
         """
 
         def _get_rooms_with_many_extremities_txn(txn):
-            sql = """
+            base_sql_pre = """
                 SELECT room_id FROM event_forward_extremities
                 GROUP BY room_id
                 HAVING count(*) > ?
+            """
+            base_sql_post = """
                 ORDER BY count(*) DESC
                 LIMIT ?
             """
+            query_args = [limit]
+            # Need if/else since 'AND room_id NOT IN ({})' fails on Postgres
+            # when len(room_id_filter) == 0. Works fine on sqlite.
+            if len(room_id_filter) > 0:
+                # questionmarks is a hack to overcome sqlite not supporting
+                # tuples in 'WHERE IN %s'
+                questionmarks = "?" * len(room_id_filter)
+
+                query_args.extend(room_id_filter)
+
+                sql = (
+                    base_sql_pre
+                    + """ AND room_id NOT IN ({})""".format(",".join(questionmarks))
```

Comment:
we normally write this as

```python
where_clause = "room_id NOT IN (%s)" % (
                ",".join("?" for _ in room_id_filter),
            )
```

also it's a pretty common idiom, so I don't think the comments at line 216 and 219 are required.